.github/workflows/aws_deploy.yml

```diff
@@ -59,6 +63,9 @@ jobs:
     runs-on: ubuntu-latest
     environment: ${{ inputs.environment }}
     needs: push_docker_image
+    if: |
+      always() &&
```

Comment:
`always()` was required in order to ensure the logic in the line below worked. I think this is slightly unintuitive, and I found several threads on the internet discussing this with lots of other people being confused by it! I did test it thoroughly though and found this to be the correct setup - it will run if and only if the previous step succeeded or was skipped.